### ISSUE TITLE
Update function can kill all table

### ISSUE THREAD

**Issue by AlexMarlo** [envms/fluentpdo#134]
Hello, we in our project use fluentpdo and we find out some problem with update function:

``` php
...
class FluentPDO {
... 
      public function update($table, $set = array(), $primaryKey = null) {
        $query = new UpdateQuery($this, $table);
        $query->set($set);
        if ($primaryKey) {
            $primaryKeyName = $this->getStructure()->getPrimaryKey($table);
            $query = $query->where($primaryKeyName, $primaryKey);
        }
        return $query;
    }

...
```

We had a lot of entities in our project wich calls update like this:

``` php
    $query = $this->_builder->update( $this->_table, $params, $user->getId());
```

but in some cases when entities not constructed properly and had null id this code cause update for all db instead of firing error, so now we replaced all such calls with safe ones for such cases:

``` php
    $query = $this->_builder->update( $this->_table, $params);
    $query->where( "id", $user->getId());
```

Our suggestions is to remove this functionality as dangerous.

**Comment by magnus-eriksson** [comment]
I disagree. I actually use this function when I need to update all records.
I don't think FluentPDO should validate your data for you. It's your task to validate the data you push to the database and I think that your comment "when entities not constructed properly" indicates that the problem isn't FluentPDO.

**Comment by AlexMarlo** [comment]
FluentPDO is a library wich should provide clear interface to use without any dangerous possibilities.
In this case i see one function wich can do both update one row or all and ", $primaryKey = null" actually sais as that this two functions in one. IMHO for more clear interface is better to separate this functions.

**Comment by magnus-eriksson** [comment]
I don't see how the interface is unclear? It states "Create UPDATE query" on that function and the primaryKey is optional.

I still argue that it is your responsibility to validate your data. If $user->getId() is null, don't do the update. FluentPDO is a fluent query builder, not an ORM.

**Comment by AlexMarlo** [comment]
https://en.wikipedia.org/wiki/SOLID_(object-oriented_design)

"Interface segregation principle
“many client-specific interfaces are better than one general-purpose interface.”"

**Comment by maldoinc** [comment]
On one hand I disagree that the library should validate the data for you, but on the other hand

the `update` shortcut must be consistent with the `update->where` call chain, so there's that;

**Comment by cbornhoft** [comment]
Closing, noted for future feature request.

**Comment by cbornhoft** [comment]
The new requirement will be that an `update()` call must be followed by at least one `where()`. If you really want to update every row on a table, `$fpdo->update(...)->where(true);` will be supported in 2.0

**Comment by AlexMarlo** [comment]
Ok, this looks as good idea.

**Comment by cbornhoft** [comment]
Fixed in 2.0
